src/MSIdentityScaffolding/Microsoft.DotNet.MSIdentity/CodeReaderWriter/DocumentBuilder.cs

```diff
@@ -0,0 +1,533 @@
+                    if (!string.IsNullOrEmpty(methodName) &&
+                        methodChanges != null &&
+                        methodChanges.CodeChanges != null)
+                    {
+                        //get method node from ClassDeclarationSyntax
+                        IDictionary<string, string>? parameterValues = null;
+                        var methodNode = modifiedClassDeclarationSyntax?
+                            .DescendantNodes()
+                            .Where(
+                                node => node is MethodDeclarationSyntax mds &&
+                                mds.Identifier.ValueText.Equals(methodName) &&
+                                (parameterValues = VerfiyParameters(methodChanges.Parameters, mds.ParameterList.Parameters.ToList())) != null)
+                            .FirstOrDefault();
+
+                        //get method's BlockSyntax
+                        var blockSyntaxNode = methodNode?.DescendantNodes().OfType<BlockSyntax>().FirstOrDefault();
+                        var modifiedBlockSyntaxNode = methodNode?.DescendantNodes().OfType<BlockSyntax>().FirstOrDefault();
+
+                        //do all the CodeChanges for the method.
+                        foreach (var change in methodChanges.CodeChanges)
+                        {
+                            if (!string.IsNullOrEmpty(change.Block) && modifiedBlockSyntaxNode != null && parameterValues != null)
+                            {
+                                //CodeChange.Parent and CodeChange.Type go together.
+                                if (!string.IsNullOrEmpty(change.Parent) && !string.IsNullOrEmpty(change.Type))
+                                {
+                                    modifiedBlockSyntaxNode = AddCodeSnippetOnParent(change, modifiedBlockSyntaxNode, parameterValues);
+                                }
+
+                                //if there is no CodeChange.Parent, check if to InsertAfter a statement.
+                                else if (!string.IsNullOrEmpty(change.InsertAfter))
+                                {
+                                    modifiedBlockSyntaxNode = AddCodeSnippetAfterNode(change, modifiedBlockSyntaxNode, parameterValues);
+                                }
+                                //if there are no Parent or InsertAfter in CodeChange,
+                                //just insert statement at the end of the block
+                                else
+                                {
+                                    modifiedBlockSyntaxNode = AddCodeSnippetInPlace(change, modifiedBlockSyntaxNode, parameterValues);
+                                }
+                            }
+                        }
+                        //replace the BlockSyntax of a MethodDeclarationSyntax of a ClassDeclarationSyntax
+                        if (blockSyntaxNode != null && modifiedBlockSyntaxNode != null && modifiedClassDeclarationSyntax != null)
```

Comment:
>blockSyntaxNode != null [](http://example.com/codeflow?start=28&length=23)

this one earlier too , if it is null the loop on line 144 is not needed

Reply:
fixed and moved to #1593